### PR TITLE
Copter: support MAV_CMD_DO_SEND_BANNER

### PR DESCRIPTION
**Pull request by jschall**

### CODE REVIEW

File: libraries/GCS_MAVLink/include/mavlink/v1.0/ardupilotmega/ardupilotmega.h [hunk 1]
```diff
@@ -109,7 +109,8 @@ typedef enum MAV_CMD
 	MAV_CMD_DO_START_MAG_CAL=42424, /* Initiate a magnetometer calibration |uint8_t bitmask of magnetometers (0 means all)| Automatically retry on failure (0=no retry, 1=retry).| Save without user input (0=require input, 1=autosave).| Delay (seconds)| Autoreboot (0=user reboot, 1=autoreboot)| Empty| Empty|  */
 	MAV_CMD_DO_ACCEPT_MAG_CAL=42425, /* Initiate a magnetometer calibration |uint8_t bitmask of magnetometers (0 means all)| Empty| Empty| Empty| Empty| Empty| Empty|  */
 	MAV_CMD_DO_CANCEL_MAG_CAL=42426, /* Cancel a running magnetometer calibration |uint8_t bitmask of magnetometers (0 means all)| Empty| Empty| Empty| Empty| Empty| Empty|  */
-	MAV_CMD_ENUM_END=42427, /*  | */
+	MAV_CMD_DO_SEND_BANNER=42428, /* Reply with the version banner |Empty| Empty| Empty| Empty| Empty| Empty| Empty|  */
+	MAV_CMD_ENUM_END=42429, /*  | */
```
Comment (magicrub) [inline libraries/GCS_MAVLink/include/mavlink/v1.0/ardupilotmega/ardupilotmega.h]: Why is this incremented by 2? You're skipping 42427.

**Comment by jschall** [comment]
I am sure it is in use on ardupilot-solo. I started using values in the 42000s in order to keep Solo stuff from conflicting with upstream stuff.

**Comment by jschall** [comment]
Nope. Possibly it was planned to be in use at the time this patch was written.

**Comment by squilter** [comment]
Ardupilot-master already supports sending a `MAV_CMD_REQUEST_AUTOPILOT_CAPABILITIES`.  It will respond with an [AUTOPILOT_VERSION](http://mavlink.org/messages/common#AUTOPILOT_VERSION).  This seems to provide pretty much the same functionality.

**Comment by rmackay9** [comment]
One issue we've discussed about using the AUTOPILOT_VERSION is the inability to fit the entire githash into the flight_custom_version, middleware_custom_version and os_custom_version fields but the suggestion there was to instead use just a small part of each of those hashes (perhaps the first 7 or 8 characters).

**Comment by jschall** [comment]
It isn't the full git hash, it's the serial number for the STM32. Bit of an oversight in the AUTOPILOT_VERSION message. It seemed risky to pick a fraction of the 12 bytes in the serial number because we couldn't figure out which bytes were most significant and which were least etc.

**Comment by jschall** [comment]
This topic has really been done to death on 3DR's issue tracker and between Tridge and I and this command was the result. Abusing other fields in AUTOPILOT_VERSION isn't a good solution... Anyway, at this point it is needed for Solo.

**Comment by rmackay9** [comment]
Merged!
As @jschall says, he and @tridge apparently discussed this at length and so in it goes!

A few minor issues:
1. the generate was done with an old version of generate so nearly every mavlink related file was changed.  I've fixed this but It's good to check through the individual commits before pushing the PR.
2. after the generate, normally we separate the version update from the .h file changes.
3. Tridge would prefer if we reduced the duplication of code related to sending the PX4 and frame strings.  I had a quick peek at doing this but it's slightly complicated because part of it is also sent along with the parameter, what's sent is different per vehicle making it awkward to move to the GCS_MAVLink library.

**Pull request closed without merging**